### PR TITLE
Moved tracked methods out of tensor

### PR DESCRIPTION
**Pull request by dmakoviichuk-tt**
### Ticket
None

### Problem description
Need to simplify GraphTracker usage in tensor.cpp.
Two steps:
1) moving methods out.
2) register them. And remove direct GraphTracekr calls.

### What's changed
Just moved them to new file.

### Checklist
- [X] Post commit CI passes
- [X] Blackhole Post commit (if applicable)
- [X] Model regression CI testing passes (if applicable)
- [X] New/Existing tests provide coverage for changes

https://github.com/tenstorrent/tt-metal/actions/runs/10568742917

### CODE REVIEW

**Review by ayerofieiev-tt — APPROVED** [review]

**Comment by ayerofieiev-tt** [comment]
@dmakoviichuk-tt please link jobs here
Screenshot 2024-08-26 at 7 24 13 PM

File: ttnn/cpp/ttnn/tensor/tensor.cpp [hunk 1]
```diff
@@ -8,31 +8,20 @@
...
 #include "ttnn/tensor/tensor_utils.hpp"
 #include "ttnn/tensor/types.hpp"
+#include "ttnn/tensor/tensor_ops.hpp"
 #include "tt_metal/common/constants.hpp"
 #include "tt_metal/common/math.hpp"
 #include "tt_metal/third_party/tracy/public/tracy/Tracy.hpp"
 #include "tt_metal/graph/graph_tracking.hpp"
 #include "ttnn/core.hpp"
-
+#include "ttnn/tensor/tensor_ops.hpp"
```
Comment (cfjchu) [inline ttnn/cpp/ttnn/tensor/tensor.cpp:22]: duplicate includes?

**Review by cfjchu — APPROVED** [review]
great!

File: ttnn/cpp/ttnn/tensor/tensor.cpp (hunk 1 above)
Comment (dmakoviichuk-tt) [reply ttnn/cpp/ttnn/tensor/tensor.cpp:22]: it was triplication :D

File: ttnn/cpp/ttnn/tensor/tensor_ops.hpp [hunk 2]
```diff
@@ -0,0 +1,46 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <vector>
```
Comment (yan-zaretskiy) [inline ttnn/cpp/ttnn/tensor/tensor_ops.hpp:7]: You don't even need this header here :-)

Comment (dmakoviichuk-tt) [reply ttnn/cpp/ttnn/tensor/tensor_ops.hpp]: Fixed :)

**Pull request merged**